### ISSUE TITLE
s_client connect doesn't send SNI server_name header through proxy by default

### ISSUE THREAD

**Issue by gmckeown** [openssl/openssl#17232]
When performing a `-connect` command using `s_client`, the default behaviour seems to be to pass the `server_name` SNI extension header, e.g.:

```$ openssl s_client -connect targetserver:443```

I verified this by packet-capture and can see the extension data in the Client Hello packet. However, when I use an HTTP proxy, the header is not sent -- verified by checking the packet and also by the target server issuing an `unrecognized name` response:

```
$ openssl s_client -connect targetserver:443 -proxy myproxy:3128

CONNECTED(00000003)
140012895368512:error:14094458:SSL routines:ssl3_read_bytes:tlsv1 unrecognized name:../ssl/record/rec_layer_s3.c:1543:SSL alert number 112
---
```

This occurs using a Squid 4.10 proxy, and a corporate proxy that I _think_ is a Blue Coat.

Verified this behaviour personally with openssl versions 1.0.2k-fips (RHEL 7.5) and 1.1.1f  (Ubuntu 20.04.3). Others have reproduced the issue with 1.1.1h as well (see https://serverfault.com/questions/1085654/sni-header-missing-from-tls-client-hello-when-using-http-proxy).

**Comment by mattcaswell** [comment]
Fix for this in #17248 (master & 3.0) and #17249 (1.1.1)

**Comment by mattcaswell** [comment]
The fix for this was merged some while ago, so closing this.
